test/e2e/e2e_test.go

```diff
@@ -1317,6 +1338,90 @@ func TestE2E(t *testing.T) {
+		if err != nil {
+			t.Fatalf("could not create test namespace: %v", err)
+		}
+		defer cleanup()
+
+		// names other than the one we need are never published to
+		neverPublished := &v1.ConfigMap{
+			TypeMeta: metav1.TypeMeta{},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:        "test-configmap-" + randSeq(5),
+				Annotations: map[string]string{api.VulnerableLegacyInjectCABundleAnnotationName: "true"},
+			},
+		}
+		_, err = adminClient.CoreV1().ConfigMaps(ns.Name).Create(context.TODO(), neverPublished, metav1.CreateOptions{})
+		if err != nil {
+			t.Fatal(err)
+		}
+		// with this name, content should never be published.  We wait ten seconds
+		err = pollForConfigMapCAInjection(adminClient, neverPublished.Name, ns.Name)
+		if err != wait.ErrWaitTimeout {
+			t.Fatal(err)
+		}
+
+		publishedConfigMap := &v1.ConfigMap{
+			TypeMeta: metav1.TypeMeta{},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:        "openshift-service-ca.crt",
+				Annotations: map[string]string{api.VulnerableLegacyInjectCABundleAnnotationName: "true"},
+			},
+		}
+		publishedConfigMap, err = adminClient.CoreV1().ConfigMaps(ns.Name).Create(context.TODO(), publishedConfigMap, metav1.CreateOptions{})
+		// tolerate "already exists" to handle the case where we're running the e2e on a cluster that already has this
+		// configmap present and injected.
+		if err != nil && !errors.IsAlreadyExists(err) {
+			t.Fatal(err)
+		}
+		publishedConfigMap, err = adminClient.CoreV1().ConfigMaps(ns.Name).Get(context.TODO(), "openshift-service-ca.crt", metav1.GetOptions{})
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// this one should be injected
+		err = pollForConfigMapCAInjection(adminClient, publishedConfigMap.Name, ns.Name)
+		if err != nil {
+			t.Fatal(err)
+		}
+		originalContent := publishedConfigMap.Data[api.InjectionDataKey]
+
+		_, hasNewStyleAnnotation := publishedConfigMap.Annotations[api.InjectCABundleAnnotationName]
+		if hasNewStyleAnnotation {
+			// add old injection to be sure only new is honored
+			publishedConfigMap.Annotations[api.VulnerableLegacyInjectCABundleAnnotationName] = "true"
+			publishedConfigMap, err = adminClient.CoreV1().ConfigMaps(ns.Name).Update(context.TODO(), publishedConfigMap, metav1.UpdateOptions{})
+			if err != nil {
+				t.Fatal(err)
+			}
+		} else {
+			// hand-off to new injector
+			publishedConfigMap.Annotations[api.InjectCABundleAnnotationName] = "true"
+			publishedConfigMap, err = adminClient.CoreV1().ConfigMaps(ns.Name).Update(context.TODO(), publishedConfigMap, metav1.UpdateOptions{})
+			if err != nil {
+				t.Fatal(err)
+			}
+		}
+
+		// the content should now change pretty quick.  We sleep because it's easier than writing a new poll and I'm pressed for time
+		time.Sleep(5 * time.Second)
+		publishedConfigMap, err = adminClient.CoreV1().ConfigMaps(ns.Name).Get(context.TODO(), publishedConfigMap.Name, metav1.GetOptions{})
+
+		// if we changed the injection, we should see different content
```

Comment:
nit: Maybe move this comment to the second block or remove it entirely? It doesn't appear relevant to the contents of the first block, since in that case different content is not expected.